addons/payment/models/payment_acquirer.py

```diff
@@ -303,6 +303,13 @@ def _get_compatible_acquirers(
 
         # Handle partner country
         partner = self.env['res.partner'].browse(partner_id)
+
+        # Partners can only make transactions for the company they belong.
+        # No acquirers should be returned when there is a mismatch of the
+        # partner's company and the company recipient of the transaction.
+        if partner.company_id.id != company_id:
```

Comment:
```suggestion
        if partner.company_id and partner.company_id.id != company_id:
```

Partners can be shared between companies (`company_id` field is left empty in that case).
If the partner record is shared, it can be used in multiple companies ;).

I believe this should solve the problems you got with the tests.

Reply:
Thank you, done :)